src/main/java/uk/ac/ebi/eva/pipeline/io/readers/NonAnnotatedVariantsMongoReader.java

```diff
@@ -31,39 +41,73 @@
  * {@link org.springframework.batch.item.data.MongoItemReader} is using
  * pagination and it is slow with large collections
  */
-public class NonAnnotatedVariantsMongoReader extends MongoDbCursorItemReader {
+public class NonAnnotatedVariantsMongoReader implements ItemStreamReader<VariantWrapper>, InitializingBean {
```

Comment:
Have you checked what state is stored in the context after making this an `ItemStreamReader`? I don't think it will be even possible to keep the cursor in a consistent state if an error occurs and the application is shut down.